operators/src/opencl/cl_program.rs

```diff
@@ -0,0 +1,578 @@
+use crate::error;
+use crate::util::Result;
+use geoengine_datatypes::call_generic_raster2d;
+use geoengine_datatypes::raster::{
+    DynamicRasterDataType, Pixel, Raster2D, RasterDataType, TypedRaster2D,
+};
+use ocl::builders::{ContextBuilder, KernelBuilder, ProgramBuilder};
+use ocl::{Buffer, Device, Kernel, MemFlags, Platform, Queue};
+use snafu::ensure;
+use snafu::ResultExt;
+
+/// Whether the kernel iterates over pixels or features
```

Comment:
this is used to determine the global work size of the kernel. For rasters it is the number of pixels in the output raster and for features it would be the number of features. I don't think there will be a case where the work size is determined by the amount of attributes.

Reply:
what about a sum per attribute?

Reply:
this would either require a 2d-kernel or custom kernel code that iterates over features but computes all the sums at once for each feature. but what would you do with the ouput? sounds like a plot output to me and goes beyond the scope of this PR I feel

Reply:
i thing it is also a tuple of some sort. How does postgres handle this?

Reply:
what does postgres have to do with opencl?

Reply:
was ich meinte war: eine Summe über eine Spalte muss man ja irgendwie abbilden. Ich denke nicht, das z.b. Postgres eienn Typ "Plot" hat. Deshalb war meine Frage wie man so etwas abbilden würde. Es hat nichts mit OpenCL zutun...

Reply:
you can computate aggregates over the whole table of via `GROUP BY` and get a new table with the sum as a new column. we could return a new DataCollection in this case